envi/archs/arm/__init__.py

```diff
@@ -9,6 +9,14 @@
 from envi.archs.arm.disasm import *
 
 class ArmModule(envi.ArchitectureModule):
+    '''
+    Following line before function is to help with different architecture
+       versions. These are not planned to be fully implemented until after ARMv8
+       is implimented. Will be important for including ARMv7M and
+       some other variants that fork from normal instruction path.
+    '''
+    archVersion = ('ARMv4')
```

Comment:
Just needed a default value. It can be easily changed but the structure is not complete yet. 
Once the CONSTANT stuff is complete I will update this and the code that uses it. 
Date: Fri, 3 Jun 2016 11:00:09 -0700
From: notifications@github.com
To: vivisect@noreply.github.com
CC: redheadedrod@hotmail.com; author@noreply.github.com
Subject: Re: [atlas0fd00m/vivisect] Update unit tests for opcodes (#3)

In envi/archs/arm/__init__.py:

> @@ -9,6 +9,14 @@
>  from envi.archs.arm.disasm import *
>  
>  class ArmModule(envi.ArchitectureModule):
> +    '''
> +    Following line before function is to help with different architecture
> +       versions. These are not planned to be fully implemented until after ARMv8
> +       is implimented. Will be important for including ARMv7M and
> +       some other variants that fork from normal instruction path.
> +    '''
> +    archVersion = ('ARMv4')

this is a problem.  we're targeting ARMv7+


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.